common/src/main/java/com/github/twitch4j/common/pool/TransactionalSubscriber.java

```diff
@@ -0,0 +1,29 @@
+package com.github.twitch4j.common.pool;
+
+/**
+ * Generic interface for objects that can respond to a subscription request, S,
+ * with a response, T, which can later be used for unsubscription to yield U.
+ *
+ * @param <S> subscription request
+ * @param <T> transactional response to the subscription request that can be used as an unsubscription request
+ * @param <U> unsubscription response
+ */
+public interface TransactionalSubscriber<S, T, U> {
```

Comment:
Is this name appropriate? It may evoke the concept of database transactions that doesn't exactly apply here